### PR TITLE
snap: add `snap help` command

### PR DESCRIPTION
**Pull request by mvo5**

### CODE REVIEW

File: cmd/snap/cmd_help.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+type cmdHelp struct{}
+
+func init() {
+	addCommand("help", shortHelpHelp, longHelpHelp, func() flags.Commander { return &cmdHelp{} })
+}
+
+var errHelp = errors.New("need-help")
+
+func (cmdHelp) Execute([]string) error {
+	return errHelp
```
Comment (pedronis) [inline cmd/snap/cmd_help.go]: couldn't this just be return flags.ErrHelp and simplify main.go? am I missing something?

**Comment by mvo5** [comment]
Fwiw, I filed a bug upstream https://github.com/jessevdk/go-flags/issues/168

**Comment by pedronis** [comment]
LGTM

**Comment by zyga** [comment]
+1

**Pull request merged**